__test__/student.spec.js

```diff
@@ -19,133 +20,395 @@ afterAll(async () => {
+      {
+        senderEmail: user3.email,
+        receiverEmail: user2.email
+      }
+    )
       .expect(200)
-      .expect((res) => {
-        assert.ok(res.text.includes(expected))
+      .then((res) => {
+        assert.deepEqual(res.body.senderEmail, user3.email)
+        assert.deepEqual(res.body.receiverEmail, user2.email)
+        FriendRequest2 = res.body;
       })
   });
 
-  it("student login test ", async () => {
+  it("Cancelling wrong request", async () => {
+    await request.post('/friend/cancel-request').send(
+      {
+        senderEmail: user2.email,
+        requestID: FriendRequest._id
+      }
+    )
+      .expect(400)
+      .then((res) => {
+        assert.deepEqual(res.body, { "errors": ["Not the sender of the request"] })
+      })
+  });
 
-    let expected = "test45"
+  it("Answering wrong request", async () => {
+    await request.post('/friend/answerFriendRequest').send(
+      {
+        receiverEmail: user1.email,
+        requestID: FriendRequest._id,
+        answer: 'accepted'
+      }
+    )
+      .expect(400)
+      .then((res) => {
+        assert.deepEqual(res.body, { "errors": ["Not the receiver of the request"] })
+      })
+  });
 
-    await request.post('/login').send(
+  it("Answering request", async () => {
+    await request.post('/friend/answerFriendRequest').send(
       {
-        "email": "ram@b.ca",
-        "password": "scooby"
+        receiverEmail: user2.email,
+        requestID: FriendRequest._id,
+        answer: 'accepted'
+      }
+    )
+      .expect(200)
+      .then((res) => {
+        assert.deepEqual(res.body, 'Friend request with ' + user1.email + ' accepted.')
       })
+    await request.post('/friend/answerFriendRequest').send(
+      {
+        receiverEmail: user2.email,
+        requestID: FriendRequest2._id,
+        answer: 'accepted'
+      }
+    )
+      .expect(200)
+      .then((res) => {
+        assert.deepEqual(res.body, 'Friend request with ' + user3.email + ' accepted.')
+      })
+  });
 
-      .expect((res) => {
-        assert.ok(res.text.includes(expected))
+  it("Verifying friends", async () => {
+    await request.get('/student/email/' + user1.email)
+      .expect(200)
+      .then((res) => {
+        assert.deepEqual(res.body.email, user1.email)
+        assert.deepEqual(res.body.friends, [user2.email])
+      })
+    await request.get('/student/email/' + user3.email)
+      .expect(200)
+      .then((res) => {
+        assert.deepEqual(res.body.email, user3.email)
+        assert.deepEqual(res.body.friends, [user2.email])
+      })
+    await request.get('/student/email/' + user2.email)
+      .expect(200)
+      .then((res) => {
+        assert.deepEqual(res.body.email, user2.email)
+        assert.deepEqual(res.body.friends, [user1.email, user3.email])
       })
   });
 
-  //works as cleanup of previous test too.
-  it("delete a Student ", async () => {
+  //Delete student tests
+  it("Delete a Student ", async () => {
```

Comment:
After deleting a student (user1.email in this case), we should test that this student is also removed from the friend list of other students (user2) in this case.

Reply:
Thus functionality does not exist. It has to be implemented first to test it

Reply:
Noted, thanks for letting me know.